pkg/scorecard/statement.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2025 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Subject {
+// 	 Repo           jsonRepoV2          `json:"repo"`
+// }
+
+package scorecard
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	intoto "github.com/in-toto/attestation/go/v1"
+
+	docs "github.com/ossf/scorecard/v5/docs/checks"
+	sce "github.com/ossf/scorecard/v5/errors"
+)
+
+const (
+	PredicateType = "https://scorecard.dev/Result/v1"
+)
+
+type Statement struct {
+	intoto.Statement
+	Predicate Predicate `json:"predicate"`
+}
+
+// Predicate overrides JSONScorecardResultV2 with a nullable Repo field.
+type Predicate struct {
+	Repo *jsonRepoV2 `json:"repo,omitempty"`
+	JSONScorecardResultV2
+}
+
+// AsStatement converts the results as an int-toto statement.
+func (r *Result) AsStatement(writer io.Writer, checkDocs docs.Doc, opt *AsJSON2ResultOption) error {
+	// Attestatio subject
+	subject := intoto.ResourceDescriptor{
+		Name: r.Repo.Name,
+		Digest: map[string]string{
+			"sha1": r.Repo.CommitSHA,
```

Comment:
in-toto/attestation supports `gitCommit` so that's an option here. https://github.com/in-toto/attestation/blob/main/spec/v1/digest_set.md#gitcommit-gittree-gitblob-gittag

Reply:
Done :+1: